h/activity/views.py

```diff
@@ -24,14 +25,20 @@ def search(request):
 
     results = []
     total = None
+    tags = []
     if 'q' in request.params:
-        query = parser.parse(request.params['q'])
+        search_query = parser.parse(request.params['q'])
 
-        out = search_lib.search(request, query)
-        total = out['total']
+        search_request = search_lib.Search(request)
+        search_request.append_filter(query.TopLevelAnnotationsFilter())
+        search_request.append_aggregation(query.TagsAggregation())
+        result = search_request.run(search_query)
+        total = result.total
+
+        tags = [t['key'] for t in result.aggregations['tags']['buckets']]
```

Comment:
Instead of this I could also see a solution where the `TagsAggregation` class has a `parse` function which would take the aggregation response and turn it into something more meaningful, in this case just a list of strings.